internal/providers/docker/components/container/component.go

```diff
@@ -17,7 +18,14 @@ type Container struct {
 type Spec compose.Service
 
 type State struct {
-	ImageID     string `json:"imageId"`
-	ContainerID string `json:"containerId"`
-	Running     bool   `json:"running"`
+	ContainerID string     `json:"containerId"`
+	Running     bool       `json:"running"`
+	Image       ImageProps `json:"image"`
+}
+
+type ImageProps struct {
```

Comment:
I've been intentionally avoiding the word `Props` in these names for two reasons: 1) in react/ui world props ~ specs and 2) in terraform, pulumi, etc, props = inputs + outputs.

A better name here might be `ImageState`?